app/models/user_content_report.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class UserContentReport
+  def initialize(reported_content:)
+    unless [Comment, RequestForComment].include?(reported_content.class)
+      raise("#{reported_content.model_name} is not configured for spam reports.")
+    end
+
+    @reported_content = reported_content
+  end
+
+  # NOTE: This implementation assumes the course URL is static and does not vary per user.
+  # This is currently valid for a majority of use cases. However, in dynamic scenarios (such as
+  # content trees in openHPI used in conjunction with A/B/n testing) this assumption may no
+  # longer hold true.
+  def course_url = lti_parameters['launch_presentation_return_url']
+
+  def human_model_name = reported_content.model_name.human
+
+  def reported_message
+    case reported_content
+      when RequestForComment
+        reported_content.question
+      when Comment
+        reported_content.text
+    end
+  end
+
+  def related_request_for_comment
+    case reported_content
+      when RequestForComment
+        reported_content
+      when Comment
+        RequestForComment.find_by!(file: reported_content.file)
```

Comment:
```suggestion
        reported_content.request_for_comment
```

Reply:
Thanks. Just to persist the knowledge here, too: The previous version had a major bug, which would cause issues with multi-file RfCs:

- The `file` attribute of RfCs indicates the active file a learner was working on when requesting help through an RfC.
- The `file` attribute of a comment refers to the file the comment belongs to.

In a scenario with two or more files, the comment being reported and the active file the learner worked on could be different. In this case, the `find_by!` would raise an exception, since no file would be found.